Automattic/newspack-plugin
# Send user metadata to AC

### All Submissions:






### Changes proposed in this Pull Request:



Contact adding/updating (w/ metadata) to ActiveCampaign when registering, updating newsletter preferences, and donating.

Partial of https://github.com/Automattic/newspack-popups/issues/833

### How to test the changes in this Pull Request:

1. Switch Newspack Newsletters to `feat/contact-add-filter` branch (https://github.com/Automattic/newspack-newsletters/pull/897)
1. Set up Newsletters to use Active Campaign ESP
2. Subscribe to the newsletter(s) using the Newsletters plugin subscription block
3. Observe the user added in AC, along with metadata:

<img width="646" alt="image" src="https://user-images.githubusercontent.com/7383192/180982132-9b51f9ae-4182-4180-80fd-c0a37d53cb7e.png">

5. Test with Registration Block (email and Google methods), Signup Block, and Donate block (using Stripe). In all cases supplying an email should result in adding a contact w/ metadata in AC.  

When donating, the following metadata fields should be added as well:

<img width="317" alt="image" src="https://user-images.githubusercontent.com/7383192/181020274-825e10d3-4155-40b5-86fb-08bbaf799394.png">

All fields handled in this PR:

- `NP_Account` – WP user ID
- `NP_Registration Date` – only if registered _without_ signing up for a list
- `NP_Newsletter Signup Date` – only if registered _and_ signed up at the same time
- `NP_Newsletter Selection` – a comma-separated list of lists the user signed up to
- `NP_Signup page`
- `NP_Signup UTM: <source|medium|campaign>` – UTM params, if present in URL
- `NP_Last Payment Date`
- `NP_Last Payment Amount`
- `NP_Billing Cycle`
- `NP_Recurring Payment`

### Other information:
